src/platform/published/pal.h

```diff
@@ -218,18 +234,18 @@ extern "C"
 #else
     enum xlang_result
```

Comment:
Seems confusing that there both com_results and xlang_results.

Reply:
To be clear, I'm not saying there's no merit for this just that we may want to bury the HRESULT in the ABI and just stick with a handful of int32_t values where absolutely necessary (e.g. QI).

Reply:
This is a side effect of the long conversation we had about reducing & simplifying the xlang error codes to a new set of values that don't align with the COM values. It's useful to note, then that the ABI for the first three methods is actually COM-compatible including failure values. It's a little verbose but nicely explicit.

Reply:
The reason for this change was to avoid confusion on what error is returned here.  We have an xlang error for no interface and an COM error for no interface, both of which have different values.  To keep compat with COM for IUnknown, we made the choice to keep this as a COM error.  If we were to represent that as an int32, it can introduce confusion to a developer on what to return here as the first instinct would be to return / check for the xlang version of no interface as we are in xlang.  But, they have to actually return the COM version of the error.  To make that clear, the int32 was changed to com_interop_result.  I do agree that it is a little verbose, but this should really only affect QueryInterface and related functions.

Reply:
Sure, I just think that if anyone is actually implementing QueryInterface and thus having to figure this out they're either one of us or we've failed because nobody should be dealing with the ABI directly.

Reply:
I agree no one else should probably be dealing with the ABI directly.  This change also makes sure we ourselves don't accidentally use the wrong error value when we return from the com interop functions by essentially having a compile time safety check.  In addition, in cross platform scenarios, it ensures that we don't introduce a new com error without explicitly being aware we are adding a new error for interop scenarios and providing a mapping to an xlang error for it.

In theory, we can make returning success less verbose by having a define S_OK or something similar as com_interop_result::success and returning that instead.  What do you think about that?

Reply:
Macros are evil. Let's leave it as is. :shipit: